test/format/test_speedscope.py

```diff
@@ -42,4 +150,228 @@ def test_speedscope_full_metrics():
+        for field in _SPEEDSCOPE_FRAME_FIELDS:
+            assert field in sframe
+
+    assert sframe_list[0]["name"] == "foo"
+    assert sframe_list[0]["file"] == "foo_module.py"
+    assert sframe_list[0]["line"] == 10
+
+    sprofile_list = speedscope_data["profiles"]
+    assert len(sprofile_list) == 4
+    for sprofile in sprofile_list:
+        for field in _SPEEDSCOPE_PROFILE_FIELDS:
+            assert field in sprofile
+        assert sprofile["type"] == "sampled"
+
+    # See the comments in the test above for a discussion of why
+    # Speedscope profiles appear in the order tested below.
+    assert sprofile_list[0]["name"] == "CPU time profile for 42:123"
+    assert sprofile_list[0]["endValue"] == 10
+    assert sprofile_list[0]["unit"] == "microseconds"
+
+    assert len(sprofile_list[0]["samples"]) == 1
+    assert sprofile_list[0]["samples"] == [[0]]
+    assert len(sprofile_list[0]["weights"]) == 1
+    assert sprofile_list[0]["weights"] == [10]
+
+    assert sprofile_list[1]["name"] == "Wall time profile for 42:123"
+    assert sprofile_list[1]["endValue"] == 20
+    assert sprofile_list[1]["unit"] == "microseconds"
+
+    assert len(sprofile_list[1]["samples"]) == 2
+    assert sprofile_list[1]["samples"] == [[0], [0]]
+    assert len(sprofile_list[1]["weights"]) == 2
+    assert sprofile_list[1]["weights"] == [10, 10]
+
+    assert sprofile_list[2]["name"] == "Memory deallocation profile for 42:123"
+    assert sprofile_list[2]["endValue"] == 30
+    assert sprofile_list[2]["unit"] == "bytes"
+
+    assert len(sprofile_list[2]["samples"]) == 1
+    assert sprofile_list[2]["samples"] == [[0]]
+    assert len(sprofile_list[2]["weights"]) == 1
+    assert sprofile_list[2]["weights"] == [30]
+
+    assert sprofile_list[3]["name"] == "Memory allocation profile for 42:123"
+    assert sprofile_list[3]["endValue"] == 20
+    assert sprofile_list[3]["unit"] == "bytes"
+
+    assert len(sprofile_list[3]["samples"]) == 1
+    assert sprofile_list[3]["samples"] == [[0]]
+    assert len(sprofile_list[3]["weights"]) == 1
+    assert sprofile_list[3]["weights"] == [20]
+
+
+def test_speedscope_wall_metrics_only():
+    with AustinFileReader(DATA_FILE) as austin:
+        mode = austin.metadata["mode"]
+        assert Mode.from_metadata(mode) == Mode.WALL
+
+        speedscope = Speedscope("austin.out", mode)
+
+        for line in austin:
+            try:
+                speedscope.add_samples(Sample.parse(line, MetricType.from_mode(mode)))
+            except InvalidSample:
+                continue
+
+        text_stream = io.StringIO()
+        speedscope.dump(text_stream)
+
+        with open(join(dirname(DATA_FILE), "austin.json"), "r") as sprof:
+            assert text_stream.getvalue() == sprof.read()
+
+    speedscope_data = speedscope.asdict()
+    for file_field in _SPEEDSCOPE_FILE_FIELDS:
+        assert file_field in speedscope_data
+
+    assert speedscope_data["$schema"] == _SPEEDSCOPE_SCHEMA_URL
```

Comment:
:+1:  This is great but I wonder if this logic could be simplified once the `austin.json` file is regenerated from the current `austin.out`.

Reply:
As far as I understand it, I've already committed a newer version of `austin.json` that shouldn't require regenerating again for this PR.

Had I regenerated the JSON before adding the hardcoded tests, I probably wouldn't have caught the error re: `cpu` vs `wall` metrics because the only discrepancies found between the prior `austin.json` and the JSON generated by the code (sans `cpu` vs `wall` fix) would have been due to the change in the `unit` field (re: microseconds), file paths, process IDs, and thread IDs. As a diagnostic, comparing against the contents of `austin.json` will tell you whether the JSON output from the code has changed for regression testing, but if the `austin.json` file itself has errors in it, then "passing" that test may be a false negative.

In the context of this project, I don't have a particularly a strong opinion on adding correctness tests to `test_speedscope_wall_metrics_only`, and I'm happy to remove those tests, because even with scripts to lessen the effort costs of data entry, it's still a substantial amount of work to extract the correct values.

Reply:
Removed in [`ac5da98f`](https://github.com/P403n1x87/austin-python/pull/7/commits/ac5da98ff3cbff3445b89e66e39ff1d4f07af3ef).